page_analyzer/utils/db.py

```diff
@@ -68,10 +63,75 @@ def create_url(url_name: str) -> int:
     return int(url_id)
 
 
-def get_urls_data() -> List[int | str]:
+def get_urls_data() -> List[Dict[str, Any]]:
+    conn = connect_db()
+    with conn.cursor(cursor_factory=psycopg2.extras.DictCursor) as dict_cur:
+        dict_cur.execute(
+            "SELECT DISTINCT ON (u.id)"
+            " u.id AS url_id,"
+            " u.name AS url_name,"
+            " c.created_at AS created_at,"
+            " c.status_code AS status_code"
+            " FROM urls u"
+            " LEFT JOIN checks c"
+            " ON c.url_id = u.id"
+            " ORDER BY u.id DESC, c.created_at DESC;"
+        )
+        urls_data_list = dict_cur.fetchall()
+    conn.close()
+    actual_urls_data = []
+    for url_data in urls_data_list:
+        actual_urls_data.append(url_data)
+    return actual_urls_data
+
+
+def generate_check_context() -> Dict[str, int | str]:
+    status_code = 200
+    h1 = ""
+    title = ""
+    description = ""
+    check_context: dict[str, int | str] = {
+        "response_code": status_code,
+        "h1": h1,
+        "title": title,
+        "description": description,
+    }
+    return check_context
+
+
+def create_check(url_id: int) -> None:
+    check_context = generate_check_context()
     conn = connect_db()
     with conn.cursor() as cur:
-        cur.execute("SELECT * FROM urls ORDER BY id DESC;")
-        urls_data_list = cur.fetchall()
+        cur.execute(
+            "INSERT INTO checks"
+            " (url_id, status_code, h1, title, description)"
+            " VALUES (%s, %s, %s, %s, %s);",
```

Comment:
postgresql should assign `None` values automatically for nullable fields that weren't used in insert operation